packages/generators/generators/lib/files/ts/plugin/tsconfig.json

```
@@ -1,24 +1,19 @@
 {
-  "extends": "@strapi/typescript-utils/tsconfigs/server",
+  "extends": "@strapi/typescript-utils/tsconfigs/admin",
 
   "compilerOptions": {
-    "outDir": "dist",
-    "rootDir": "."
+    "target": "ESNext",
+    "strict": true
   },
 
-  "include": [
-    // Include the root directory
-    "server",
-    // Force the JSON files in the src folder to be included
-    "server/**/*.json"
-  ],
+  "include": ["admin", "custom.d.ts"],
 
   "exclude": [
     "node_modules/",
     "dist/",
 
     // Do not include admin files in the server compilation
```

I think this comment may need to be updated or removed 

```suggestion
    // Do not include server files in the server compilation
```